app/views/contexts/summary_context.py

```diff
@@ -94,7 +94,9 @@ def build_summary_item(
         )["summary"]
 
         if summary.get("sections"):
-            self.summaries.extend(summary["sections"])
+            for section in summary["sections"]:
+                if any(group["blocks"] for group in section["groups"]):
+                    self.summaries.extend(summary["sections"])
```

Comment:
The indentation can be removed by using `for section in summary.get("sections", []):`

Is it possible, some groups might not have block?

Reply:
Yeah that's why we were getting some of the empty sections, for example in my test schema the household section was generating groups with empty blocks

Reply:
Updated the above 👍

Reply:
The check if does any groups have a block, is it possibe, some group will have block and some won't. In that case, you would still get empty blocks?. Or is that check in templates?

Reply:
Yeah that is guarded against in the template